Tests/FluentAssertions.Equivalency.Specs/DataTableSpecs.cs

```diff
@@ -32,6 +32,105 @@ public void When_data_tables_are_both_null_equivalence_test_should_succeed()
         ((DataTable)null).Should().BeEquivalentTo(null);
     }
 
+    [Fact]
+    public void When_row_match_mode_is_invalid_it_should_fail()
+    {
+        // Arrange
+        var typedDataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+        var typedDataSet2 = new TypedDataSetSubclass(typedDataSet1);
+
+        var dataTable1 = typedDataSet1.ToUntypedDataSet().Tables["TypedDataTable1"];
+        var dataTable2 = typedDataSet2.ToUntypedDataSet().Tables["TypedDataTable1"];
+
+        // Act
+        Action action = () => dataTable1.Should().BeEquivalentTo(dataTable2, options => options.UsingRowMatchMode((RowMatchMode)2));
+
+        // Assert
+        action.Should().Throw<XunitException>().WithMessage(
+            "Unknown RowMatchMode *when trying to compare *");
+    }
+
+    [Theory]
+    [MemberData(nameof(EmptyPrimaryKeys))]
+    public void When_row_match_mode_is_primary_key_without_primary_key_it_should_fail(DataColumn[] emptyPrimaryKey)
+    {
+        // Arrange
+        var typedDataSet1 = CreateDummyDataSet<TypedDataSetSubclass>(includeRelation: false);
```

Comment:
🤔 Do you really need two datasets here? If not, remove one to not obfuscate the test. If you do need it, please use subject and expectation as the variable names.